src/java.base/share/classes/jdk/internal/foreign/SystemLookup.java

```diff
@@ -80,16 +85,41 @@ private static SymbolLookup makeSystemLookup() {
         }
     }
 
+    /* Obtain the address of the inlined function list by loading the specified library. */
+    private static MemorySegment getInlinedFunctListAddr() {
+         MemorySegment funcsAddr = null;
+
+        /* Directly load the specified library only once with the inlined libc functions
+         * missing in the default library.
+         */
+        SymbolLookup funcsLibLookup = libLookup(libs -> libs.load(jdkLibraryPath("syslookup")));
+
+        if (OperatingSystem.isAix()) {
+            funcsAddr = funcsLibLookup.find("funcs").orElseThrow().reinterpret(AixFuncSymbols.LAYOUT.byteSize());
+        } else {
+            /* Perform a FFI downcall to obtain the address of the inlined function list on z/OS. */
+            MemorySegment funcs_addr = funcsLibLookup.find("funcs_addr").orElseThrow();
+            MethodHandle handle = Linker.nativeLinker().downcallHandle(funcs_addr, FunctionDescriptor.of(ADDRESS));
+            try {
+                funcsAddr = ((MemorySegment)handle.invoke()).reinterpret(ZosFuncSymbols.LAYOUT.byteSize());
+            } catch (Throwable e) {
+                throw new InternalError(e);
+            }
+        }
+
+        return funcsAddr;
+    }
+
     /* The method is to wrap up the SymbolLookup address generated by forward-porting
      * the default library loading solution introduced in Java 16, which remains valid
-     * on AIX.
+     * on AIX or z/OS.
      */
-    private static SymbolLookup makeAixLookup() {
+    private static SymbolLookup makeDefaultLookup() {
         NativeLibrary lib = NativeLibraries.defaultLibrary;
         return name -> {
             Objects.requireNonNull(name);
             MemorySegment funcAddr = null;
-            AixFuncSymbols symbol = AixFuncSymbols.valueOfOrNull(name);
+            var symbol = (OperatingSystem.isAix()) ? AixFuncSymbols.valueOfOrNull(name) : ZosFuncSymbols.valueOfOrNull(name);
```

Comment:
Please remove the unnecessary parentheses around `OperatingSystem.isAix()`.

Reply:
Fixed.